mongofiles/options.go

```diff
@@ -65,16 +64,9 @@ func ParseOptions(rawArgs []string) (Options, error) {
 	opts.WriteConcern = wc
 
 	// set ReadPreference
-	if inputOpts.ReadPreference != "" {
-		opts.ReadPreference, err = db.ParseReadPreference(inputOpts.ReadPreference)
-		if err != nil {
-			return Options{}, fmt.Errorf("error parsing --readPreference: %v", err)
-		}
-	} else {
-		opts.ReadPreference, err = readpref.New(readpref.NearestMode)
-		if err != nil {
-			return Options{}, fmt.Errorf("error setting default read preference: %v", err)
-		}
+	opts.ReadPreference, err = db.ParseReadPreference(inputOpts.ReadPreference, opts.URI.ParsedConnString())
+	if err != nil {
+		return Options{}, fmt.Errorf("error parsing --readPreference: %v", err)
```

Comment:
Because an error could occur as a result of parsing the URI now, I think this error (and similar ones) should be updated to just say "error parsing read preference...", omitting the "--".